src/model/element-base.ts

```diff
@@ -95,7 +131,180 @@ export abstract class ElementBase implements Feature {
    * single list. A mixin can be applied more than once, each time its
    * members override those before it in the prototype chain.
    */
-  mixins: Reference[] = [];
+  readonly mixins: ImmutableArray<Reference>;
+
+  constructor(init: ElementBaseInit, document: Document) {
+    const {
+      description,
+      summary,
+      demos,
+      events,
+      sourceRange,
+      properties,
+      attributes,
+      methods,
+      astNode,
+      warnings,
+      slots,
+      privacy,
+      jsdoc,
+      superClass,
+      mixins,
+    } = init;
+    this.description = description;
+    this.summary = summary;
+    this.demos = demos ? Array.from(demos) : [];
+    this.sourceRange = sourceRange;
+    this.astNode = astNode;
+    this.warnings = warnings ? Array.from(warnings) : [];
+    this.slots = slots ? Array.from(slots) : [];
+    this.privacy = privacy;
+    this.jsdoc = jsdoc;
+    this.superClass = superClass ? superClass.resolve(document) : undefined;
+    this.mixins = (mixins || []).map((m) => m.resolve(document));
+
+    const prototypeChain = this._getPrototypeChain(document, init);
```

Comment:
made this after the comment below: this is a little of a confusing name, because you're not getting the whole prototype chain, you're combining the superclasses and mixins. It might be more obvious to just inline this function, or move all inheritance into this function under a different name